core/src/main/scala/akka/kafka/KafkaConsumerActor.scala

```diff
@@ -160,11 +161,14 @@ private[kafka] class KafkaConsumerActor[K, V](settings: ConsumerSettings[K, V])
 
   override def preStart(): Unit = {
     super.preStart()
+
     consumer = settings.createKafkaConsumer()
+    currentPollTask = schedulePollTask()
   }
 
   override def postStop(): Unit = {
-    pollTask.cancel()
+    if (currentPollTask != null)
+      currentPollTask.cancel()
```

Comment:
The problem with this solution is that there is a small (but existing) chance that there is already a `Poll` message in flight in the actor's mailbox when the actor is **restarted**, and that can double the frequency of the polls.

One way to solve that is to include a `target: KafkaConsumer` in the `Poll` message and when receiving `Poll` ignore any `target != this`.